pkg/tunnel/healthz.go

```diff
@@ -0,0 +1,62 @@
+package tunnel
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/paralus/paralus/pkg/grpc"
+	rpcv3 "github.com/paralus/paralus/proto/rpc/scheduler"
+	commonv3 "github.com/paralus/paralus/proto/types/commonpb/v3"
+	infrav3 "github.com/paralus/paralus/proto/types/infrapb/v3"
+	"github.com/paralus/relay/pkg/utils"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func syncClusterHeath(sni string, status commonv3.ParalusConditionStatus, reason string) {
+	id, err := getClusterID(sni)
+	if err != nil {
+		_log.Error("unable to get clusterID", "error", err)
+	}
```

Comment:
Should we return early if unable to get cluster id?

Reply:
I would like cluster healthz to be non-impacting core functionality. We may not cluster cluster id due to few reasons like service unavailable at that moment etc. All core functionality including retries in baked into relay. Updating cluster health should not impact the core flow and hence just logging them and continuing.

Reply:
Understood your point, added a return to error cases.